packages/mui-utils/src/isFocusVisible/isFocusVisible.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Returns a boolean indicating if the event's target has :focus-visible
+ */
+export default function isFocusVisible(element: Element): boolean {
```

Comment:
I would have expected this to come from Base UI.

?
```suggestion
// TODO import from Base UI
```

Reply:
As mentioned in the description: "ideally, we would use the `:focus-visible` pseudo-selector directly, but that would mean removing the `focusVisible` classes and refactoring many components."

The `isFocusVisible` function is an intermediary step, so I don't think we should implement it in Base UI. When we reimplement the look of the library, we should use `:focus-visible` directly, deprecate the `focusVisible` classes, and eventually remove this function.

Reply:
This is not how I understand the logic. 50% of the use of `isFocusVisible` is to apply the focusVisible class name, the other 50% is to change JavaScript behavior. I don't see how this can go away.

Reply:
> The other 50% is to change React states or JavaScript behaviors

From the components in this PR, the only one that uses `isFocusVisible` to manage JavaScript behavior is `ButtonBase`. Two out of three of those are related to ripple pulse, which, if I recall correctly, is not present in MD3.

The remaining one of the three relates to event handling, which should eventually come from Base UI. If they don't have a use case for managing JavaScript behavior, then I don't think Material UI will. cc: @michaldudak.

I'm not sure; I haven't dug into this yet as it's not a priority at all, but we might be able to remove it eventually.

Reply:
We don't use such a utility currently in Base UI. However, there's one use case where it may help - when we focus an element programmatically during an mousedown event it sometimes incorrectly enters the focus-visible state. This happens as browsers (seemingly) wait for the click event to determine if a mouse interaction caused focus change. A custom utility could fix it.

Reply:
> Two out of three of those are related to ripple pulse, which, if I recall correctly, is not present in MD3.

@DiegoAndai If by pulse we mean: 

https://github.com/user-attachments/assets/11e3e843-d75c-449b-9bdb-846c4160738b

Agree, it's gone. I think it's a great move of MD3, didn't like that. Otherwise, ripple is still present: https://material-web.dev/components/button/.

> uses isFocusVisible to manage JavaScript behavior

What seem to be true:

- Button: can be replaced with `:focus-visible`
- Slider: can be replaced with `:focus-visible`
- Switch: can be replaced with `:focus-visible`
- Link: can be replaced with `:focus-visible`
- Rating: can be replaced with `:focus-visible`
- ButtonBase: pulse ripple, on its way out
- Tooltip: required for behavior. Calling .focus() shouldn't show the tooltip. It's also necessary in Base UI, but today hidden under https://github.com/floating-ui/floating-ui/blob/ecd627e4676a50643edcdf8915a6acefa50ee3d8/packages/react/src/hooks/useFocus.ts#L124.